### PR TITLE
feat(gtd): allow count for weekdays

### PR DESCRIPTION
**Pull request by max397574**
this allows to have a time.due/start for example for the 4th monday that will come.
The format is 4mon.

### CODE REVIEW

**Comment by danymat** [comment]
Nice, It seems that the tests passes. Would you mind adding one or two tests for the new case ?

**Comment by max397574** [comment]
I'll see what I can do

File: lua/neorg/modules/core/gtd/queries/helpers.lua [hunk 1]
```diff
@@ -44,6 +44,21 @@ module.public = {
...
         end
+        local amount, weekday = text:match("^(%d+)(%a%a%a)$")
+        if not vim.tbl_contains(vim.tbl_keys(values), weekday) then
+            weekday = nil
+        end
+
+        if amount and weekday then
+            local date = os.date("*t")
+            if values[weekday] > date.wday then
+                date.day = date.day + values[weekday] - date.wday
```
Comment (max397574) [inline lua/neorg/modules/core/gtd/queries/helpers.lua:55]: I think I need to put a `+(amount-1)*7` here right?

**Review by danymat — APPROVED** [review]

**Comment by danymat** [comment]
merged, thanks !! 🥳

**Pull request merged**